src/combineReducers.js

```diff
@@ -134,6 +134,7 @@ export default function combineReducers(reducers) {
       nextState[key] = nextStateForKey
       hasChanged = hasChanged || nextStateForKey !== previousStateForKey
     }
+    hasChanged = hasChanged || Object.keys(state).length !== finalReducerKeys.length
```

Comment:
@gaearon quite a puzzler if Object.keys is too slow. Should I fix the development case first and then think about this a bit more?